hips/hip-9999.md

```diff
@@ -0,0 +1,56 @@
+---
+hip: 9999
+title: "Pre-defined release dates for Helm"
+authors: [ "Marc Khouzam <marc.khouzam@montreal.ca>" ]
+created: "2020-09-04"
+type: "process"
+status: "draft"
+---
+
+## Abstract
+
+Have well-known cyclical release dates for Helm minor and patch releases.
+Have published release dates for Helm major releases, once such a release has entered the implementation phase.
+
+## Motivation
+
+Contributors and users/organisations greatly appreciate knowing in advance when the next release will be available.
+It allows contributors to plan their work and organisations to schedule migration plans; it also avoids disappointing delays and provides users with clear timelines for access to bug fixes and features.
+
+## Rationale
+
+Minor and patch releases for Helm normally occur a couple of times a year.  They therefore lend themselves well to cyclical, pre-defined release dates.
+
+As it is not recommended to use Helm with a version of Kubernetes that is newer than the version Helm was compiled against ([see here][helm-skew]), there is value in aligning the date for Helm minor releases with [Kubernetes releases][kubernetes-dates], which occur every 3 to 4 months.
+
+Helm major releases happen rarely and therefore do not lend themselves to pre-defined, cyclical release dates.  However, once a major release has been defined and has entered the implementation phase, a specific release date should be chosen and announced.
+
+## Specification
+
+Patch releases should be done once a month on the second Wednesday of each month.
```

Comment:
Is it really necessary to schedule patch releases at a specific date? Our previous system was to release patches on an as-needed basis. That being said, we haven't had a "perfect" .0 release in a long time.

What about a system where we discuss during dev calls whether we need to cut a patch release, and if so, we discuss a date, preferring Tuesdays (patch Tuesday) and Wednesdays whenever possible? That would match up with the current system.

Reply:
I agree that a monthly patch schedule is a bit of a burden.  And the monthly schedule may be too agressive and we could choose something less frequent

My point was really about the fixed date, where my motivation was that judging if "we need to cut a patch release" can be subjective and make it theoretically possible that a bug fix stays unreleased for a long time; fixed dates give a guarantee to contributors that their bug fix will be released within a known period of time.

We can also start with fixed dates for minor releases and see how things go before fixing patch release dates.